### DISCUSSION TITLE
How to see/log raw failed SQL queries?

### DISCUSSION THREAD

**Discussion by vitaliiznak** [prisma/prisma#20297] [Q&A] [answered]
```javascript
prisma.$on('error', (e) => {
  console.error(e)
});
```
The error object doesn't contain the query that was send for execution and `prisma.$on('query'` does not print the failed query either.

**Comment by nurul3101** [comment]
Hello @vitaliiznak 👋 

Can you share how is the  Prisma Client configured ?

Have you subscribed to the error log event?
```ts
const prisma = new PrismaClient({
  log: ['query', 'info', 'warn', 'error'],
})
``` 

The exact [event (e) type and the properties available](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#event-types)  depends on the log level.

**Reply by vitaliiznak** [reply, marked as answer]
@nurul3101 
I am closing the question
I was searching for  **UPSERT** in  SQL queries logs. I figured out that the **upsert** operation in **Prisma does not directly translate** to **SQL UPSERT** operation
The logs are present

Thank you

**Reply by nurul3101** [reply]
Thank you for getting back and letting us know. 🙂
